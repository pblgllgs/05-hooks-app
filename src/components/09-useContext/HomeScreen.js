import React from 'react'

export const HomeScreen = () => {
    return (
        <>
            <h1>Home Screen</h1>
            <hr />
        </>
    )
}
