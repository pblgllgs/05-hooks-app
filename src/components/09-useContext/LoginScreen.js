import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const userlogin = {
        id:123,
        name:'pablo',
        email: 'pablo@gmail.com'
    } 

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className='btn btn-primary' 
                onClick={()=> setUser(userlogin)}
            >
                Login
            </button>
        </div>
    )
}
