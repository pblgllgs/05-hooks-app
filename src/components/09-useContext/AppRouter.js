import React from 'react'

import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <NavBar />
                <Switch>
                    <Route exact path="/" element={<HomeScreen />} />
                    <Route exact path="/about" element={<AboutScreen />} />
                    <Route exact path="/login" element={<LoginScreen />} />
                    <Route path="*" element={<HomeScreen />} />
                </Switch>
            </Router>
        </div>
    )
}