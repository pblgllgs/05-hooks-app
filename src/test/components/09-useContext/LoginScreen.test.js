import { mount } from 'enzyme'
import { LoginScreen } from '../../../components/09-useContext/LoginScreen'
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('pruebas sobre <LoginScreen />', () => {
    const setUser = jest.fn()
    const user = {
        id: 123,
        name: 'pablo',
        email: 'pablo@gmail.com'
    } 
    const wrapper = mount(
        //se agrega un contexto para la prueba y se utiliza el valor de usuario creado
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    );
    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('debe de ejecutar el setuser con el argumento esperado', () => {
        //wrapper.find('button').prop('onClick')(user) otra forma utilizando los prop del elemnto html
        //simulamos el click y mandamos como parametro el user
        wrapper.find('button').simulate('click', { target: { user } })
        //se compara con un snapshot
        expect(wrapper).toMatchSnapshot();
        //se espera que se llame la funcion con un objeto user como parametro
        expect(setUser).toHaveBeenCalledWith(user)
    })
})
