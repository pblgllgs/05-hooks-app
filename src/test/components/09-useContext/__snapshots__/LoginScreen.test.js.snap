// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre <LoginScreen /> debe de ejecutar el setuser con el argumento esperado 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`pruebas sobre <LoginScreen /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
